src/files/NcFile.cpp

```diff
@@ -147,6 +147,8 @@ NcFile::NcFile(std::string path, File::Mode mode)
 NcFile::~NcFile() {
     auto status = nc_close(file_id_);
     assert(status == NC_NOERR);
+    // silent "unused variable" when compiling without assertions
```

Comment:
Would it be better to combine the above statement to `nc_close(file_id_) == NC_NOERR`?

Reply:
Inside the assert? No, since asserts should not have side effects. Or are you suggesting something else?